### PR TITLE
txn: fix pessimitic exist check(#19004)

### PR DESCRIPTION
**Pull request by cfzjywxk**
### What problem does this PR solve?

Cherry-pick #19004  to release 4.0

### What is changed and how it works?

What's Changed:

- Record the keys having presume not exist flag in statement context. **Pass the exist map using `txn.SetOption` interface and the exist map will be used in `unionStore.Get`.** It's different from the master branch behaviour.
- Record pessimistic lock values for both point get and non point get locks, these value will be used to check existence.
- Check existence if needed in LockKeys function for pessimistic transactions, report error if the value already exists.

How it Works:

### Check List 

Tests 

- Unit test

Side effects

### Release note 

- fix key existence check for pessimistic transactions.

### CODE REVIEW

File: store/tikv/txn.go [hunk 1]
```diff
@@ -360,10 +362,26 @@ func (txn *tikvTxn) LockKeys(ctx context.Context, lockCtx *kv.LockCtx, keysInput
 		}
 	}()
 	txn.mu.Lock()
+	logutil.Logger(ctx).Info("[for debug] key len", zap.Int("len", len(keysInput)))
```
Comment (jackysp) [inline store/tikv/txn.go:365]: should we remove this debug log?

**Comment by coocood** [comment]
LGTM

**Pull request merged**